keps/sig-apps/4026-job-creation-timestamp-annotation/README.md

```diff
@@ -0,0 +1,310 @@
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [X] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Currently, there is no supported way to get the original/expected initial scheduled timestamp for the job created from a cronjob. This KEP proposes to set the the original scheduled time as an annotation in the job metadata.
+
+## Motivation
+
+### Goals
+
+- Set job scheduled timestamp as an annotation on the job.
+- Adding the annotation should not be disruptive to existing workloads.
+
+### Non-Goals
+
+## Proposal
+
+At a high level, the proposal is to modify the CronJob controller to set the job scheduled timestamp as a job annotation. The details of this are outlined in the Design Details section below.
+
+Job scheduled timestamp annotation: `batch.kubernetes.io/cronjob-scheduled-timestamp`
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a user, I would like to get the job's scheduled timestamp that this job was expected to be running.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+One thing that must be considered is how enabling this new feature will interact with existing workloads. There are a couple of options:
+
+1. Only inject the job annotation for *newly created CronJobs*. We can track this by annotating newly created CronJobs to distinguish existing ones from newly created ones.
+Using this strategy, either none of the jobs have this annotation, or all of them do, which will provide a more consistent user experience. However, in the case of a cluster downgrade to a version without this feature, new jobs would start getting created without this annotation again.
+
+1. Inject the annotation on *all jobs* (jobs existing prior to feature enablement and jobs created after feature enablement). However, retroactively modifying jobs of existing workloads would risk being too disruptive to existing workloads which may have logic depending on job annotations, so this option should not be considered.
```

Comment:
Like I said in the other comment, I don't think that option was ever on the table. CronJobs always operate on newly created Jobs, and this feature should be identical.